src/main/java/com/onlinebookstore/onlinebookstore/controller/BookController.java

```diff
@@ -30,29 +33,35 @@ public class BookController {
 
     @GetMapping
     @Operation(summary = "Find All Books", description = "Find All Books in DB")
-    public List<BookDto> getAll(Pageable pageable) {
-        return bookService.findAll(pageable);
+    @PreAuthorize("hasRole('USER')")
+    public List<BookDto> getAll(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return bookService.findAll(user.getEmail(), pageable);
     }
 
     @GetMapping("/{id}")
     @Operation(summary = "Find Book by id", description = "Find Book by id")
+    @PreAuthorize("hasRole('USER')")

```

Comment:
Only for user? Or maybe for user and admin?

Reply:
Well i task it says SECURITY REQUIREMENTS
Available for non authenticated users:

    POST: /api/auth/register

Available for users with role USER

    GET: /api/books
    GET: /api/books/{id}

Available for users with role ADMIN

    POST: /api/books/
    PUT: /api/books/{id}
    DELETE: /api/books/{id}

So yes I thought aswell that admin should be able to get books.. but maybe its security measures?